### PR TITLE
Preliminary Turing support

### PR DESCRIPTION
**Pull request by Echelon9**
Support for initial Turing-series GPUs in envytools:
- Add preliminary support for Turing (TU102 and TU104)
- Add GDDR6 detection support (Turing)
- demmt: Support TURING_CHANNEL_GPFIFO_A
- demmt: Add preliminary TURING_DMA_COPY support

Verified support in `nvbios` against two Turing VGA BIOS files (GeForce RTX 2080 Ti and GeForce RTX 2080).

Note: This requires a fix for multi-part vbios'es, which have been seen since GP108 (e.g. @karolherbst has a potential patch).

### CODE REVIEW

File: ssot/gpu.py [hunk 1]
```diff
@@ -1251,6 +1256,26 @@ class GpuGV100(GpuGP100):
     bios_major = 0x88
     bios_chip = 0x00
 
+# Turing
+
+class GpuTU102(GpuGV100):
+    gen = GenTuring
+    id = 0x162
+    pciid = 0x1e04  # also 0x1e07
```
Comment (koriakin) [inline ssot/gpu.py]: write 0x1e00 here -- the low pciid_varbits bits are variable anyhow

File: ssot/gpu.py [hunk 2]
```diff
@@ -1251,6 +1256,26 @@ class GpuGV100(GpuGP100):
...
+    pciid_varbits = 7
+    gpc_count = 6
+    tpc_count = 6
+    bios_major = 0x90
+    bios_chip = 0x02
+
+class GpuTU104(GpuTU102):
+    id = 0x164
+    pciid = 0x1e82  # also 0x1e87
```
Comment (koriakin) [inline ssot/gpu.py]: 0x1e80

File: ssot/gpu.py (hunk 1 above)
Comment (Echelon9) [reply ssot/gpu.py]: Done

File: ssot/gpu.py (hunk 2 above)
Comment (Echelon9) [reply ssot/gpu.py]: Done

**Pull request merged**